Source/SwiftLintFramework/Rules/VerticalWhitespaceBetweenCasesRule.swift

```diff
@@ -0,0 +1,149 @@
+import Foundation
+import SourceKittenFramework
+
+private extension File {
+    func violatingRanges(for pattern: String) -> [NSRange] {
+        return match(pattern: pattern, excludingSyntaxKinds: SyntaxKind.commentAndStringKinds)
+    }
+}
+
+private extension Dictionary where Key: StringProtocol, Value: StringProtocol {
+    func cleanedKeysDict() -> [String: String] {
+        var cleanDict = [String: String]()
+
+        for (key, value) in self {
+            guard let keyString = key as? String else { continue }
+            let cleanedKey = keyString.replacingOccurrences(of: "↓", with: "")
+            cleanDict[cleanedKey] = value as? String
+        }
+
+        return cleanDict
+    }
+}
+
+public struct VerticalWhitespaceBetweenCasesRule: ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    private static let nonTriggeringExamples = [
+        """
+        switch x {
+
+        case 0..<5:
+            print("x is low")
+
+        case 5..<10:
+            print("x is high")
+
+        default:
+            print("x is invalid")
+
+        }
+        """,
+        """
+        switch x {
+        case 0..<5:
+            print("x is low")
+
+        case 5..<10:
+            print("x is high")
+
+        default:
+            print("x is invalid")
+        }
+        """,
+        """
+        switch x {
+        case 0..<5: print("x is low")
+        case 5..<10: print("x is high")
+        default: print("x is invalid")
+        }
+        """
+    ]
+
+    private static let violatingToValidExamples: [String: String] = [
+    """
+        switch x {
+        case 0..<5:
+            print("x is valid")
+    ↓    default:
+            print("x is invalid")
+        }
+    """: """
+        switch x {
+        case 0..<5:
+            print("x is valid")
+
+        default:
+            print("x is invalid")
+        }
+    """
+    ]
+
+    private let pattern = "([^\\n{][ \\t]*\\n)([ \\t]*(?:case[^\\n]+|default):[ \\t]*\\n)"
+}
+
+extension VerticalWhitespaceBetweenCasesRule: OptInRule, AutomaticTestableRule {
+    public var configurationDescription: String { return "N/A" }
```

Comment:
you shouldn't override this

Reply:
Override what?

Reply:
Overriding `configurationDescription` and `init(configuration: Any)` prevents the default violation severity configuration versions of these from working. I fixed this in this PR.